pkg/ssh/ssh.go

```diff
@@ -84,101 +42,53 @@ func (c config) Translate(u *url.URL) *url.URL {
 	return newURL
 }
 
-func (p *parser) read(fileName string) error {
-	var file io.Reader
-	if p.open == nil {
-		f, err := os.Open(fileName)
-		if err != nil {
-			return err
-		}
-		defer f.Close()
-		file = f
-	} else {
-		var err error
-		file, err = p.open(fileName)
-		if err != nil {
-			return err
-		}
+func (t *Translator) resolve(hostname string) (string, error) {
+	t.cacheMu.RLock()
+	cached, cacheFound := t.cacheMap[strings.ToLower(hostname)]
+	t.cacheMu.RUnlock()
+	if cacheFound {
+		return cached, nil
```

Comment:
As far as I can tell this cache is never actually populated in this method. I am wondering about its purpose though. We can rectify this and start populating the cache or I think it would be okay to just allow the users of this package to implement the caching since our implementation is just an in-memory map. What are your thoughts?

Reply:
Good catch. If the cache is not populated, that was an oversight on my part.

I do think that an in-memory cache would be beneficial since `Translate()` is typically used to translate many URLs at once: think of a git repo that has multiple SSH remotes—Translate will be called 2 times per each remote. Since most remotes will be to the same hostname, a built-in cache avoids shelling out to `ssh` repeatedly.